src/deploy/Network.ts

```diff
@@ -77,6 +77,16 @@ export async function deployNetworkComet(
     );
   }
 
+  const protocolConfiguration = {
+    governor: timelock ? timelock.address : ethers.constants.AddressZero,
+    pauseGuardian: timelock ? timelock.address : ethers.constants.AddressZero,
+    ...await getConfiguration(deploymentManager, contractMapOverride, configurationOverrides),
+  };
+
+  // Validate that the configuration is properly set before trying to deploy contracts
+  // XXX awkward that this has to be here instead of in the beginning of the function
```

Comment:
We can't move this validation to the beginning right now because the code paths for deploying `SimpleTimelock` and `GovernorSimple` need to be executed before we can generate `protocolConfiguration`. I'm thinking about moving the deployment of `SimpleTimelock` and `GovernorSimple` out of `deployNetworkComet` and into each testnet's own `deploy.ts` since they are not production contracts.

Will tackle this refactor in a separate PR though.

Reply:
This will actually be tackled by Jared in his refactoring/idempotent deploys PR.